src/Components/App/App.js

```diff
@@ -0,0 +1,56 @@
+import React, {useState, useEffect} from 'react';
+import './App.css';
+import ColorContainer from '../ColorContainer/ColorContainer';
+import SavePaletteForm from '../SavePaletteForm/SavePaletteForm';
+
+const App = () => {
+
+  let [paletteColor_1, setPaletteColor_1] = useState('#000000');
+  let [paletteColor_2, setPaletteColor_2] = useState('#bbbbbb');
+  let [paletteColor_3, setPaletteColor_3] = useState('#222222');
+  let [paletteColor_4, setPaletteColor_4] = useState('#dddddd');
+  let [paletteColor_5, setPaletteColor_5] = useState('#444444');
+
+  let [isSaved_1, setIsSaved_1] = useState(false);
+  let [isSaved_2, setIsSaved_2] = useState(false);
+  let [isSaved_3, setIsSaved_3] = useState(false);
+  let [isSaved_4, setIsSaved_4] = useState(false);
+  let [isSaved_5, setIsSaved_5] = useState(false);
+
+  const generateRandomColors = () => {
+    if (!isSaved_1) {
+      setPaletteColor_1('#'+Math.floor(Math.random()*16777215).toString(16))
+    };
+    if (!isSaved_2) {
+      setPaletteColor_2('#'+Math.floor(Math.random()*16777215).toString(16))
+    };
+    if (!isSaved_3) {
+      setPaletteColor_3('#'+Math.floor(Math.random()*16777215).toString(16))
+    };
+    if (!isSaved_4) {
+      setPaletteColor_4('#'+Math.floor(Math.random()*16777215).toString(16))
+    };
+    if (!isSaved_5) {
+      setPaletteColor_5('#'+Math.floor(Math.random()*16777215).toString(16))
+    };
+  }
+
+  useEffect(() => generateRandomColors(), []);
```

Comment:
This replaces componentDidMount right? So each time a color changes, state changes, triggering re-render as I understand